src/pkg/shub/client/util.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package client
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+)
+
+// isShubPullRef returns true if the provided string is a valid Shub
+// reference for a pull operation.
+func isShubPullRef(shubRef string) bool {
+	//define regex for each URI component
+	registryRegexp := `([-.a-zA-Z0-9/]{1,64}\/)?` //target is very open, outside registry
+	nameRegexp := `([-a-zA-Z0-9]{1,39}\/)`        //target valid github usernames
+	containerRegexp := `([-_.a-zA-Z0-9]{1,64})`   //target valid github repo names
+	tagRegexp := `(:[-_.a-zA-Z0-9]{1,64})?`       //target is very open, file extensions or branch names
+	digestRegexp := `(\@[a-f0-9]{32})?`           //target file md5 has, git commit hash, git branch
```

Comment:
The branch or commit hash will not match this regex. An unabbreviated SHA1 hash is `{40}` and a branch name is much more open.

Reply:
I'll set it for SHA1 for now